quill-core/src/main/scala/io/getquill/norm/ApplyIntermediateMap.scala

```diff
@@ -12,6 +14,8 @@ object ApplyIntermediateMap {
       case Filter(Map(a: GroupBy, b, c), d, e)    => None
       case SortBy(Map(a: GroupBy, b, c), d, e, f) => None
       case Map(a: GroupBy, b, c) if (b == c)      => None
+      case Map(Distinct(Map(a, b, c)), d, e) if isomorphic(e, c, d) =>
```

Comment:
- add a test to `ApplyIntermediateMapSpec`
- add the transformation to the end with a comment similar to the other transformations